lib/mutant/expression/method.rb

```diff
@@ -3,44 +3,50 @@ class Expression
 
     # Explicit method expression
     class Method < self
+      include Anima.new(:scope_name, :scope_symbol, :method_name)
+      private(*anima.attribute_names)
 
       MATCHERS = IceNine.deep_freeze(
         '.' => Matcher::Methods::Singleton,
         '#' => Matcher::Methods::Instance
       )
 
-      register(
-        /\A(?<scope_name>#{SCOPE_PATTERN})(?<scope_symbol>[.#])(?<method_name>#{METHOD_NAME_PATTERN})\z/
-      )
+      METHOD_NAME_PATTERN = Regexp.union(
+        /(?<method_name>[A-Za-z_][A-Za-z\d_]*[!?=]?)/,
```

Comment:
> Do you plan on supporting unicode method names in the future?

Yeah, but thats more complex than I initially thought.

> Is the plan for right now to just restrict this as much as possible and in a future PR expand this to include all valid ruby method names?

My plan is to implement expression parsing semantically equivalent to the previous implementation, but fully mutation tested.

I plan a followup that allows all valid literal ruby method names parsed as expressions. Please note that through the changed inner workings generating expressions from matched subjects already works with any valid ruby method names, as the need to parse expressions from subjects was removed.